### PR TITLE
[STACK 1] Add passing mimic tests

### PR DESCRIPTION
**Pull request by SimonBiggs**
Some tests required some bug fixes. These are the tests that didn't only needed very minor library changes.

### CODE REVIEW

File: lib/pymedphys/tests/mosaiq/test_mimicked_db.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+PATIENT_ID = 989898
+FIELD_ID = 88043
+A_TREATMENT_DATETIME = "2020-04-27 08:03:28.513"
+MACHINE_ID = "2619"
+FIELD_NAME = "3ABUT"
+TIMEZONE = "Australia/Sydney"
+FIRST_NAME = "MOCK"
+LAST_NAME = "PHYSICS"
+FULL_NAME = f"{LAST_NAME}, {FIRST_NAME.capitalize()}"
```
Comment (sjswerdloff) [inline lib/pymedphys/tests/mosaiq/test_mimicked_db.py:34]: Why not capitalize the surname?

File: lib/pymedphys/tests/mosaiq/test_mimicked_db.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    data_paths = pymedphys.zip_data_paths("metersetmap-gui-e2e-data.zip")
+    date = A_TREATMENT_DATETIME.split(" ")[0]
+
+    filtered_paths = [
+        path
+        for path in data_paths
+        if date in str(path) and str(FIELD_ID) in str(path) and path.suffix == ".trf"
+    ]
+
+    return filtered_paths[0]
```
Comment (sjswerdloff) [inline lib/pymedphys/tests/mosaiq/test_mimicked_db.py:58]: This assumes ordering in the zip is explicitly what you need or can accept.
Do you have a preferred first item / ordering?

File: lib/pymedphys/tests/mosaiq/test_mimicked_db.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        path for path in data_paths if path.name == f"{PATIENT_ID}_{FIELD_NAME}.dcm"
+    ]
+
+    return filtered_paths[0]
+
+
+@pytest.mark.mosaiqdb
+def test_get_patient_name(connection):
+    name = helpers.get_patient_name(connection, PATIENT_ID)
+    assert name == FULL_NAME
```
Comment (sjswerdloff) [inline lib/pymedphys/tests/mosaiq/test_mimicked_db.py:75]: Is it worth doing a capitalization insensitive compare of some kind?

File: lib/pymedphys/tests/mosaiq/test_mimicked_db.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+    treatments = helpers.get_treatments(connection, start, end, MACHINE_ID)
+    assert np.datetime64(A_TREATMENT_DATETIME) in treatments["start"].tolist()
+
+
+@pytest.mark.mosaiqdb
+def test_trf_identification(connection: pymedphys.mosaiq.Connection, trf_filepath):
+    delivery_details = pymedphys.beta.trf.identify(connection, trf_filepath, TIMEZONE)
+    assert delivery_details.field_id == FIELD_ID
+    assert delivery_details.first_name == FIRST_NAME
```
Comment (sjswerdloff) [inline lib/pymedphys/tests/mosaiq/test_mimicked_db.py:105]: Is capitalization important?

**Review by sjswerdloff — APPROVED** [review]

File: lib/pymedphys/tests/mosaiq/test_mimicked_db.py (hunk 1 above)
Comment (SimonBiggs) [reply lib/pymedphys/tests/mosaiq/test_mimicked_db.py:34]: Weirdly "capitalize" actually means make the first letter be capital and the rest be lower case.

So in this case FULL_NAME will be "PHYSICS, Mock".

This essentially a choice based on the current API.

File: lib/pymedphys/tests/mosaiq/test_mimicked_db.py (hunk 2 above)
Comment (SimonBiggs) [reply lib/pymedphys/tests/mosaiq/test_mimicked_db.py:58]: Yeah, I do actually. Some of those TRF files are partial deliveries. Only the first one is a full delivery. Potentially I should be explicitly pulling the first one by name. I'll do that I think.

Cheers

File: lib/pymedphys/tests/mosaiq/test_mimicked_db.py (hunk 3 above)
Comment (SimonBiggs) [reply lib/pymedphys/tests/mosaiq/test_mimicked_db.py:75]: Potentially, it's just the API at the moment of that function by definition is trying to "pretty print" the name. So, it's sort of testing that as well?

File: lib/pymedphys/tests/mosaiq/test_mimicked_db.py (hunk 4 above)
Comment (SimonBiggs) [reply lib/pymedphys/tests/mosaiq/test_mimicked_db.py:105]: See above.

File: lib/pymedphys/tests/mosaiq/test_mimicked_db.py (hunk 2 above)
Comment (SimonBiggs) [reply lib/pymedphys/tests/mosaiq/test_mimicked_db.py:54]: Thanks for this @sjswerdloff. Fixed in c33c7f7c46070ca4ae5e48758cbd8787f7074940

File: lib/pymedphys/tests/mosaiq/test_mimicked_db.py (hunk 4 above)
Comment (SimonBiggs) [reply lib/pymedphys/tests/mosaiq/test_mimicked_db.py]: Fixed in 43edbc618acd60f55eae27d30944342b18b9fb4f

File: lib/pymedphys/tests/mosaiq/test_mimicked_db.py (hunk 3 above)
Comment (SimonBiggs) [reply lib/pymedphys/tests/mosaiq/test_mimicked_db.py:71]: I'll leave this one, as (at the moment) that specific case is a result of the `helpers.get_patient_name` API.

**Comment by SimonBiggs** [comment]
Thanks for this review @sjswerdloff 🙂. Given this is approved I'll merge in. Let me know if you have any further recommendations stemming from the above adjustments based on your recommendations.

**Pull request merged**